UI: Display an error message and disable salt actions when salt is unavailable

JBWatenbergScality opened the pull request:
**Component**:

UI

**Context**: 

Instead of login the user out in case of salt API unavailability, the user should be prompted with a message disclaiming it and be able to access monitoring screens. Salt related actions should be disabled.

Closes: #3117

JBWatenbergScality committed:
solution-navbar: Add an nginx docker image serving the navbar

JBWatenbergScality committed:
ui: Replace navbar and authentication mechanism with solution-ui-navbar

JBWatenbergScality committed:
ui: Add an error boundary in case of failure to load the navbar

JBWatenbergScality committed:
solutions-navbar: Add navigation role to the navbar

JBWatenbergScality committed:
ui: Review tests following to solutions navbar integration

Refs: #3104

JBWatenbergScality committed:
salt: Modify UI config in order to set the new oidc redirect uri correctly following to solution navbar integration

JBWatenbergScality committed:
ui: Handle salt unavailability

Refs: #3117, #3109

bert-e commented:
# Hello jbwatenbergscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/ui-salt-error-management` and the
destination branch `development/2.8`.

Please resolve the conflict on **the feature branch** (`improvement/ui-salt-error-management`).

```bash
 $ git fetch
 $ git checkout origin/improvement/ui-salt-error-management
 $ git merge origin/development/2.8
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/ui-salt-error-management
```

JBWatenbergScality committed:
solutions-navbar: Rename to shell-ui

JBWatenbergScality committed:
salt: Add build and deploy states for shell-ui

JBWatenbergScality committed:
UI : Split webpack bundle into smaller chunks by lazy loading routes components

New bundles size are as followed :
  571.11 KB  build/static/js/4.25b92d7b.chunk.js
  36.11 KB   build/static/js/15.8684c3a2.chunk.js
  30 KB      build/static/js/main.20da19d4.chunk.js
  19.76 KB   build/static/js/1.c5d9f7f7.chunk.js
  12.88 KB   build/static/css/4.023dbf36.chunk.css
  9.85 KB    build/static/js/12.e6067137.chunk.js
  6.91 KB    build/static/js/14.3bd7ce01.chunk.js
  5.62 KB    build/static/js/0.69f7df4d.chunk.js
  2.65 KB    build/static/js/6.8d8d1636.chunk.js
  2.28 KB    build/static/js/7.06fb1003.chunk.js
  1.99 KB    build/static/js/5.c144a469.chunk.js
  1.7 KB     build/static/js/10.39e394e2.chunk.js
  1.3 KB     build/static/js/runtime-main.33ce3b79.js
  1.2 KB     build/static/js/11.e2755908.chunk.js
  1009 B     build/static/js/9.a5906d8a.chunk.js
  489 B      build/static/js/8.5381719b.chunk.js
  441 B      build/static/css/main.9707ba81.chunk.css
  329 B      build/static/js/13.dc12452d.chunk.js

We should investigate in bundling core ui in chunks per components to go further because for now all core-ui components are bundled in metalk8s ui even if they are not used.

Refs: https://github.com/scality/metalk8s/issues/3124

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/ui-salt-error-management` has diverged from
`development/2.9` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.9` into `improvement/ui-salt-error-management`
* Rebase `improvement/ui-salt-error-management` onto `origin/development/2.9`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.
